src/server/components/formRunner/helpers.ts

```diff
@@ -146,8 +147,17 @@ export async function initialiseFormRunnerSession({
     `initialising form runnner session for list item id: ${listItem.id} with isAnnualReview ${isAnnualReview}`
   );
 
-  const listItemJsonData = listItem?.jsonData as ListItemJsonData;
-  listItem.jsonData = merge(listItem.jsonData, listItemJsonData.updatedJsonData ?? {});
+  // merge with listItem.jsonData
+  let jsonData = listItem.jsonData as ListItemJsonData;
+  listItem.jsonData = merge(listItem.jsonData, jsonData?.updatedJsonData ?? {});
+
+  // merge with listItem.address
+  jsonData = listItem.jsonData as ListItemJsonData;
+  listItem.address = {
+    ...listItem.address,
+    ...getChangedAddressFields(jsonData, listItem.address)
+  };
+
```

Comment:
"Temp" variables aren't really needed. I'm not sure why jsonData is pulled out so soon. 

 There is possibly some confusion around let vs const. Even though something is `const` - it just means that the reference cannot be changed. e.g. you can't do 

```
const immutableRef = { myProp: 1 }
immutableRef = foo;
```

you can do 

```
const immutableRef = { myProp: 1 };
immutableRef.myProp = 3.

```


I won't block on this specific issue any further, since it works, but my suggestion is here:


```suggestion
  const { updatedJsonData, ...jsonData } = listItem.jsonData;
  const mergedJsonData = { // lodash merge isn't necessary.
    ...jsonData,
    ...updatedJsonData // if it's undefined, it's ignored.
  };
  
  const listItemForInit = {
  ...listItem, 
  jsonData: mergedJsonData,
  address: {
    ...listItem.address,
    ...getChangedAddressFields(mergedJsonData, listItem.address)
    }
  }
  
```

Reply:
no confusion over const vs let here but there was a ts linting error when passing in listItem.jsonData.updatedJsonData directly into the merge function.  Splitting out and casting then calling updatedJsonData on the cast object worked.  Either way, I've implemented your suggestion.